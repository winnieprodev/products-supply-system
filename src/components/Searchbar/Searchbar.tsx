import { library as iconLibrary } from "@fortawesome/fontawesome-svg-core";
import { faSearch } from "@fortawesome/free-solid-svg-icons";
import { SearchIcon } from "components/Icons";
import React, { useCallback, useEffect, useRef } from "react";
import { useHistory } from "react-router";
import ROUTES from "routes";

import * as Styled from "./searchbar.styled";

iconLibrary.add(faSearch);

interface OwnProps {
  value?: string;
  onChange?: (newValue: string) => void;
  onCancel?: () => void;
  navigateToSearch?: boolean;
  placeholder?: string;
  focus?: boolean;
}

const Searchbar: React.FC<OwnProps> = ({
  value,
  onChange,
  navigateToSearch,
  onCancel,
  focus,
  placeholder = "Search for a supplier or a product",
}) => {
  const onChangeCallback = useCallback(
    (event: React.ChangeEvent<HTMLInputElement>) => {
      if (onChange) {
        onChange(event.target.value);
      }
    },
    [onChange],
  );

  const onKeydownCallback = useCallback(
    (event: React.KeyboardEvent<HTMLInputElement>) => {
      if (onCancel && event.key === "Escape") {
        onCancel();
      }
    },
    [onCancel],
  );

  const inputRef = useRef<HTMLInputElement>(null);

  useEffect(() => {
    if (focus) {
      inputRef?.current?.focus();
    }
  }, [focus, inputRef]);

  const history = useHistory();

  const onFocus = useCallback(() => {
    if (navigateToSearch) {
      history.push(ROUTES.SEARCH.use({ value: "" }));
    }
  }, [history, navigateToSearch]);
  return (
    <Styled.SearchBar>
      <Styled.Icon>
        <SearchIcon />
      </Styled.Icon>
      <Styled.Input
        ref={inputRef}
        value={value}
        onChange={onChangeCallback}
        onKeyDown={onKeydownCallback}
        placeholder={placeholder}
        onFocus={onFocus}
      />
    </Styled.SearchBar>
  );
};

export default Searchbar;
