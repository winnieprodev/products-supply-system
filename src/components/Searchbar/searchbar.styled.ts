import styled from "styled-components";
import {
  primaryColor,
  secondaryColor,
  secondaryColorLighter,
} from "styles/colors";

export const SearchBar = styled.div`
  border-radius: 10px;
  height: 30px;
  display: flex;
  align-items: center;
  background: ${secondaryColor};
  padding: 6px 12px;
  position: relative;
  transition: background ease 0.2s;

  &:hover,
  &:active {
    background: ${secondaryColorLighter};
  }
`;

export const Icon = styled.div`
  font-size: 22px;
  padding-right: 12px;
  border-right: 1px solid ${primaryColor};
  margin-right: 12px;
`;

export const Input = styled.input`
  position: absolute;
  left: 0;
  top: 0;
  bottom: 0;
  padding-left: 60px;
  width: calc(100% - 60px);
  outline: none;
  border: none;
  background: none;
`;
