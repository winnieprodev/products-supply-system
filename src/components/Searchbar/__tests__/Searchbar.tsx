import { render } from "@testing-library/react";

import Searchbar from "../Searchbar";

describe("Searchbar", () => {
  describe("Snapshot tests", () => {
    test("it matches the snapshot", () => {
      const { container } = render(
        <Searchbar value="1" onChange={jest.fn()} />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
