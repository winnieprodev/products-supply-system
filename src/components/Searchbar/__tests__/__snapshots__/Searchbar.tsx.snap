// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searchbar Snapshot tests it matches the snapshot 1`] = `
.c0 {
  border-radius: 10px;
  height: 30px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #EEEEEE;
  padding: 6px 12px;
  position: relative;
  -webkit-transition: background ease 0.2s;
  transition: background ease 0.2s;
}

.c0:hover,
.c0:active {
  background: #F3F3F3;
}

.c1 {
  font-size: 22px;
  padding-right: 12px;
  border-right: 1px solid #F25C05;
  margin-right: 12px;
}

.c2 {
  position: absolute;
  left: 0;
  top: 0;
  bottom: 0;
  padding-left: 60px;
  width: calc(100% - 60px);
  outline: none;
  border: none;
  background: none;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-search fa-w-16 "
      data-icon="search"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
        fill="currentColor"
      />
    </svg>
  </div>
  <input
    class="c2"
    placeholder="Search for a supplier or a product"
    value="1"
  />
</div>
`;
